docs/development/machine-bootstrap.md

```diff
@@ -0,0 +1,19 @@
+# Machine-Controller integration into Gardener
+The Gardener is deploying the [machine-controller-manager](https://github.com/gardener/machine-controller-manager) as part of a Shoot controlplane onto a Seed to manage the nodes/machines of the Shoot cluster. The machine-controller-manager will create, bootstrap, update and delete the nodes of the Shoot cluster and the underlying virtual machines.
+
+## Flow of creating and bootstrapping Machines/Nodes
+The gardener-controller-manager will ensure during the creation/reconciliation flow
+- that the machine-controller-manager gets deployed into the Shoot namespace of the Seed.
+- that proper ``<cloudprovider>machineclass`` objects and its corresponding secrets exists in the Shoot namespace of the Seed. Those secrets contain information to bootstrap the node and will be copied to the machines of the related class.
```

Comment:
s/its/their

Reply:
s/information to bootstrap/information how to bootstrap

Reply:
s/the node/the nodes/

(one machine-secret is responsible for the entire worker pool)

Reply:
> and will be copied to the machines of the related class.

what is meant by that? The machine controller manager configures the VMs to have the provided cloud-config/user-data specified in the machine class secret.

Reply:
s/``<cloudprovider>machineclass``/`<cloudprovider>MachineClass`